versioned/tiered/tiered-impl/src/main/java/org/projectnessie/versioned/impl/InternalBranch.java

```diff
@@ -403,57 +410,71 @@ private void save(Store store) {
      */
     private static InternalBranch collapseIntentionLog(UpdateState initialState, Store store, InternalBranch branch, int attempts)
         throws ReferenceNotFoundException, ReferenceConflictException {
-      try {
-        for (int attempt = 0; attempt < attempts; attempt++) {
-
-          // cleanup pending updates.
-          UpdateState updateState = attempt == 0 ? initialState : branch.getUpdateState(store);
-
-          // ensure that any to-be-saved items are saved. This is a noop on attempt 0 since
-          // ensureAvailable will have already done a save.
-          updateState.save(store);
-
-          // now we need to take the current list and turn it into a list of 1 item that is saved.
-          final ExpressionPath commits = ExpressionPath.builder("commits").build();
-          final ExpressionPath last = commits.toBuilder().position(updateState.finalL1position).build();
-
-          UpdateExpression update = UpdateExpression.initial();
-          ConditionExpression condition = ConditionExpression.initial();
-
-          for (Delete d : updateState.deletes) {
-            ExpressionPath path = commits.toBuilder().position(d.position).build();
-            condition = condition.and(ExpressionFunction.equals(path.toBuilder().name(ID).build(), d.id.toEntity()));
-            update = update.and(RemoveClause.of(path));
-          }
-
-          condition = condition.and(ExpressionFunction.equals(last.toBuilder().name(ID).build(),
-              updateState.finalL1RandomId.toEntity()));
-
-          // remove extra commits field for last commit.
-          update = update
-              .and(RemoveClause.of(last.toBuilder().name(Commit.DELTAS).build()))
-              .and(RemoveClause.of(last.toBuilder().name(Commit.KEY_MUTATIONS).build()))
-              .and(SetClause.equals(last.toBuilder().name(Commit.PARENT).build(), updateState.finalL1.getParentId().toEntity()))
-              .and(SetClause.equals(last.toBuilder().name(Commit.ID).build(), updateState.finalL1.getId().toEntity()));
-
-          InternalRef.Builder<?> producer = EntityType.REF.newEntityProducer();
-          boolean updated = store.update(ValueType.REF, branch.getId(), update, Optional.of(condition), Optional.of(producer));
-          if (updated) {
-            LOGGER.debug("Completed collapse update on attempt {}.", attempt);
-            return producer.build().getBranch();
+      try (Scope outerScope = createSpan("InternalBranch.collapseIntentionLog")
+          .withTag("nessie.operation", "CollapseIntentionLog")
+          .withTag("nessie.branch", branch.getName())
+          .startActive(true)) {
+        try {
+          UpdateState updateState = initialState;
+          for (int attempt = 0; attempt < attempts; attempt++) {
+            try (Scope innerScope = createSpan("Attempt-" + attempt).startActive(true)) {
```

Comment:
again minor concern over cost of this op vs payoff

Reply:
Yea, that particular one is tricky. The code is "too far away" from knowing whether the user wants things to be traced, but if you trace, the information is pretty useful.